### DISCUSSION TITLE
bearer JWT token from SendRequest to another SendRequest

### DISCUSSION THREAD

**Discussion by JesusVFR** [postmanlabs/postman-flows#281] [Q&A]
[](https://user-images.githubusercontent.com/75691077/158053731-fd47619a-acc6-4f93-b7df-b108357130d3.png)[](https://user-images.githubusercontent.com/75691077/158053906-a0b172a4-d0d3-4817-829b-7c83462fb1ba.png)Hi all,

can you help on my scenario, I have to concatenate 2 **SendRequest**

1. SentRequest -> Login , I saved the returned Login JWT token as durable variable

![image](https://user-images.githubusercontent.com/75691077/158054213-bc93ae32-223d-4d48-bfe2-db8f0ef7e608.png)

2. I want to call another "SendRequest" block, being an authenticated service, i need to pass the bearer token on headers, setting the Authorization key how can I do that ?

![image](https://user-images.githubusercontent.com/75691077/158054219-d52dfa09-2179-49d6-b696-5475448a039f.png)

**Comment by tlustopro** [comment]
Hello,

try to create two request by standart way (in collection) and add them to separate "send request" block.

**Reply by JesusVFR** [reply]
Hello,

I need to pass header parameter between the "Sendrequest" blocks, how can I achieve this with your suggestions ?

**Reply by tlustopro** [reply]
I dont know way, how to do that between flow blocks using durable variable right now.

But you can create first request named "Login" and save response for JWT token to environment variable. After that i will create next request, where you can use this variable in header.

Next step is create two blocks, where one will be "Sendrequest" for Login and another "Sendrequest" block for request where token is used.

**Reply by JesusVFR** [reply]
Ok thanks @tlustopro I'll try your suggestion.

**Reply by tlustopro** [reply]
let me kwow, if my suggestion works.

**Reply by JesusVFR** [reply]
Hi @tlustopro , it works, but with latest saved data, I mean, if I call 

1. outside the Flow:
   Call1 -> set variable -> Call2 read fresh variable, all is it working
   
2 run PostmanFlows
  Call1 -> set variable -> Call2, read the previous set variable, so in the flow, seems not reading latest fresh variable's value.

![image](https://user-images.githubusercontent.com/75691077/159229715-fb9b0922-711b-414e-8287-7be7a8e60818.png)


If I call the 2 endpoint outside the flow, works...

Any further suggestions ?

**Reply by tlustopro** [reply]
Hi @JesusVFR in your colection try to unset variable first, after that set variable again. Something like that:

`pm.collectionVariables.unset("variable_key");`
`pm.collectionVariables.set("variable_key", "variable_value");`

Do that in every call of request.

**Reply by JesusVFR** [reply]
Hi @tlustopro 

thanks for the tip, but seems not updating the variable : 

![image](https://user-images.githubusercontent.com/75691077/159452838-5c72f162-ab77-4058-a822-191c2539251b.png)


![image](https://user-images.githubusercontent.com/75691077/159452532-f5700868-350e-4eec-b63b-c93b6e9a0c38.png)


the 2nd call, **ShopAuthentication/Heartbeat** seems ready the previous set variable value.
